elyra/pipeline/processor_local.py

```diff
@@ -112,17 +129,75 @@ def _sort_operation_dependencies(operations_by_id: dict, ordered_operations: lis
                                                                         parent_operation)
             ordered_operations.append(operation)
 
-    def _get_envs(self, operation: Operation) -> Dict:
-        """Operation stores environment variables in a list of name=value pairs, while
-           subprocess.run() requires a dictionary - so we must convert.  If no envs are
-           configured on the Operation, the existing env is returned, otherwise envs
-           configured on the Operation are overlayed on the existing env.
-        """
-        envs = os.environ.copy()
-        for nv in operation.env_vars:
-            nv_pair = nv.split("=")
-            if len(nv_pair) == 2:
-                envs[nv_pair[0]] = nv_pair[1]
-            else:
-                self.log.warning(f"Could not process environment variable entry `{nv}`, skipping...")
-        return envs
+
+class OperationProcessor(ABC):
+
+    def __init__(self):
+        self.log = log.get_logger()
+
+    @abstractmethod
+    def operation_name(self) -> str:
+        pass
+
+    @abstractmethod
+    async def process(self, operation: Operation, filepath: str):
+        pass
+
+
+class NotebookOperationProcessor(OperationProcessor):
+    _operation_name = 'execute-notebook-node'
+
+    @property
+    def operation_name(self) -> str:
+        return self._operation_name
+
+    async def process(self, operation: Operation, filepath: str):
+        if not os.path.isfile(filepath):
+            raise FileNotFoundError(f'Could not find {filepath}')
+
+        notebook_dir = os.path.dirname(filepath)
+        notebook_name = os.path.basename(filepath)
+
+        self.log.debug(f'Processing: {filepath}')
+
+        argv = ['papermill', filepath, filepath, '--cwd', notebook_dir]
+        envs = operation.env_vars_as_dict
+        t0 = time.time()
+        try:
+            subprocess.run(argv, env=envs, check=True)
+        except Exception as ex:
+            self.log.error(f'Internal error executing {filepath}')
+            raise RuntimeError(f'Internal error executing {filepath}') from ex
+
+        t1 = time.time()
+        duration = (t1 - t0)
+        self.log.debug(f'Execution of {notebook_name} took {duration:.3f} secs.')
+
+
+class PythonScriptOperationProcessor(OperationProcessor):
+    __operation_name = 'execute-python-node'
+
+    def operation_name(self) -> str:
+        return self.__operation_name
+
+    async def process(self, operation: Operation, filepath: str):
```

Comment:
These methods differ by exactly one line.  Seems like the `process()` method itself should be on the ABC and let a member variable (e.g., `self.argv` be configured by each subclass.

All the other comments apply here as well.